configs/example/gem5_library/x86-mi300x-gpu.py

```diff
@@ -0,0 +1,155 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""
+Script to run a full system GPU simulation.
+
+Usage:
+------
+```
+scons build/VEGA_X86/gem5.opt
+./build/VEGA_X86/gem5.opt
+    configs/example/gem5_library/x86-viper-gpu.py
+    --image <disk image>
+    --kernel <kernel>
+    --app <gpu application>
+```
+
+Example:
+--------
+```
+./build/VEGA_X86/gem5.opt
+    configs/example/gem5_library/x86-viper-gpu.py
+    --image ./gem5-resources/src/x86-ubuntu-gpu-ml/disk-image/x86-ubuntu-gpu-ml
+    --kernel ./gem5-resources/src/x86-ubuntu-gpu-ml/vmlinux-gpu-ml
+    --app ./gem5-resources/src/gpu/square/bin.default/square.default
+```
+"""
+
+import argparse
+
+from gem5.coherence_protocol import CoherenceProtocol
+from gem5.components.devices.gpus.amdgpu import MI300X
+from gem5.components.memory.single_channel import SingleChannelDDR4_2400
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.components.processors.simple_processor import SimpleProcessor
+from gem5.isas import ISA
+from gem5.prebuilt.viper.board import ViperBoard
+from gem5.prebuilt.viper.cpu_cache_hierarchy import ViperCPUCacheHierarchy
+from gem5.resources.resource import (
+    DiskImageResource,
+    FileResource,
+)
+from gem5.simulate.simulator import Simulator
+from gem5.utils.requires import requires
+
+requires(
+    isa_required=ISA.X86,
+    coherence_protocol_required=CoherenceProtocol.GPU_VIPER,
+)
+
+# Kernel, disk, and applications are obtained locally.
+parser = argparse.ArgumentParser()
+
+parser.add_argument(
+    "--image",
+    type=str,
+    required=True,
+    help="Full path to the gem5-resources x86-ubuntu-gpu-ml disk-image.",
+)
+
+parser.add_argument(
+    "--kernel",
+    type=str,
+    required=True,
+    help="Full path to the gem5-resources vmlinux-gpu-ml kernel.",
+)
+
+parser.add_argument(
+    "--app",
+    type=str,
+    required=True,
+    help="Path to GPU application, python script, or bash script to run",
+)
+
+parser.add_argument(
+    "--kvm-perf",
+    default=False,
+    action="store_true",
+    help="Use KVM perf counters to give accurate GPU insts/cycles with KVM",
+)
+
+args = parser.parse_args()
+
+# stdlib only supports up to 3GiB currently. This will need to be expanded in
+# the future.
+memory = SingleChannelDDR4_2400(size="3GiB")
```

Comment:
I haven't tried Pannotia.  The 3GiB constraint is specific to stdlib I think because interleaved memory is required. It has worked for about 15 years in the deprecated configs, which get around the problem but adding a non-interleaved memory range.

Reply:
FYI: we are working on fixing this here at UC Davis